platform/mellanox/mlnx-sfpd/scripts/mlnx-sfpd

```diff
@@ -98,37 +100,78 @@ class MlnxSfpd:
     def initialize(self):
         self.state_db.connect("STATE_DB")
 
-        # open SDK API handle
-        # retry at most SX_OPEN_RETRIES times to wait
-        # until SDK is started during system startup
-        retry = 1
-        while True:
-            rc, self.handle = sx_api_open(None)
-            if rc == SX_STATUS_SUCCESS:
-                break
-
-            log_warning("failed to open SDK API handle... retrying {}".format(retry))
+        swid_cnt_p = None
 
-            time.sleep(2 ** retry)
-            retry += 1
-
-            if retry > self.SX_OPEN_RETRIES:
-                raise RuntimeError("failed to open SDK API handle after {} retries".format(retry))
-
-        rc = sx_api_host_ifc_open(self.handle, self.rx_fd_p)
-        if rc != SX_STATUS_SUCCESS:
-            raise RuntimeError("sx_api_host_ifc_open exited with error, rc {}".format(rc))
+        try:
+            # Wait for SDK daemon to be started with detect the sdk_ready file
+            retry = 0
+            while not os.path.exists(SDK_DAEMON_READY_FILE):  
```

Comment:
>while not os.path.exists(SDK_DAEMON_READY_FILE): [](start = 12, length = 48)

Did you check the warm-reboot flow? syncd container will be killed, during next reboot, is the file there or not?